core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -148,10 +148,19 @@ class AdminClient(val time: Time,
       throw new IllegalArgumentException(s"Consumer Group $groupId with protocol type '${metadata.protocolType}' is not a valid consumer group")
 
     Errors.forCode(metadata.errorCode()).maybeThrow()
-    val consumers = metadata.members.map { consumer =>
-      val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(Utils.readBytes(consumer.memberAssignment)))
-      ConsumerSummary(consumer.memberId, consumer.clientId, consumer.clientHost, assignment.partitions.toList)
-    }.toList
+    val consumers = metadata.state match {
+      case "Stable" =>
+        metadata.members.map { consumer =>
+          val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(Utils.readBytes(consumer.memberAssignment)))
+          ConsumerSummary(consumer.memberId, consumer.clientId, consumer.clientHost, assignment.partitions.toList)
+        }.toList
+      case "PreparingRebalance" | "AwaitingSync" =>
```

Comment:
> Do you mean along with the list of consumers (with no assignment)?

Not sure I understand. I think it's reasonable to show all members with no assignment when the group is rebalancing, but it might be nice to print a warning that the group is rebalancing so users don't have to guess what's going on.